collections/src/main/scala/strawman/collection/Iterator.scala

```diff
@@ -469,18 +469,12 @@ trait Iterator[+A] extends IterableOnce[A] with IterableOnceOps[A, Iterator, Ite
         if (!self.hasNext) false
         else {
           nextElement = self.next()
-          val y = f(nextElement)
-          if (traversedValues.contains(y)) {
-            loop()
-          } else {
-            traversedValues += y
-            nextElementDefined = true
-            true
-          }
+          if (traversedValues.add(f(nextElement))) true else loop()
```

Comment:
* can be replaced with `traversedValues.add(f(nextElement)) || loop()`
* `nextElementDefined` never changes and is always `false`.

Reply:
Good suggestion for simplification, thanks!
As for `nextElementDefined` it's now changed at line 476.